framework/BaseClasses/BaseEntity.py

```diff
@@ -55,30 +55,33 @@ def getSolutionExportVariableNames(cls):
     return {}
 
   def __init__(self):
+    """
+      Construct.
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
     self.name             = ''                                                          # name of this istance (alias)
     self.type             = type(self).__name__                                         # specific type within this class
     self.verbosity        = None                                                        # verbosity level (see message handler)
     self.globalAttributes = {}                                                          # this is a dictionary that contains parameters that are set at the level of the base classes defining the types
     self._knownAttribute  = []                                                          # this is a list of strings representing the allowed attribute in the xml input for the class
     self._knownAttribute += ['name','verbosity']                                        # attributes that are known
     self.printTag         = 'BaseType'                                                  # the tag that refers to this class in all the specific printing
-    self.messageHandler   = None                                                        # message handling object
     self.variableGroups   = {}                                                          # the variables this class needs to be aware of
     self.metadataKeys     = set()                                                       # list of registered metadata keys to expect from this entity
     self.metadataParams   = {}                                                          # dictionary of registered metadata keys with repect to their indexes
 
-  def readXML(self,xmlNode,messageHandler,variableGroups={},globalAttributes=None):
+  def readXML(self, xmlNode, variableGroups=None, globalAttributes=None):
```

Comment:
Eventually that would be ideal. For now most of RAVEN is using readXML patterns instead of handleInput patterns. (I did try removing it, then saw the changes would be very large and out of the scope of this work.)